cps-pit/slice.go

```
@@ -38,9 +54,39 @@ func (slice *Slice) addDoc(doc bson.Raw) error {
 		return err
 	}
 
+	if slice.NumDocs == 1 {
+		slice.Start = rawToBsonD(doc)
+	}
+	slice.End = rawToBsonD(doc)
 	return nil
 }
 
+func rawToBsonD(doc bson.Raw) bson.D {
+	var bsonDoc bson.D
+	if err := bson.Unmarshal(doc, &bsonDoc); err != nil {
+		println("Error unmarshalling bson.Raw to bson.D", err)
+		return nil
```

This error should not be swallowed but instead be bubbled up to a slicer method and send to the `errc` chan to stop all oplog tailing.